stash_engine/app/models/stash_engine/curation_activity.rb

```diff
@@ -172,6 +174,33 @@ def email_orcid_invitations
     end
     # rubocop:enable Metrics/AbcSize
 
+    # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity
+    def update_publication_flags
+      case status
+      when 'withdrawn'
+        resource.update_columns(meta_view: false, file_view: false)
+      when 'embargoed'
+        resource.update_columns(meta_view: true, file_view: false)
+      when 'published'
+        resource.update_columns(meta_view: true, file_view: true)
+      end
+
+      return if resource&.identifier.nil?
+
+      resource.identifier.update_column(:pub_state, status)
+
+      return if %w[withdrawn embargoed].include?(status)
+
+      # find out if there were not file changes since last publication and reset file_view, if so.
+      unchanged = true
+      resource.identifier.resources.reverse_each do |res|
+        break if res.id != resource.id && res&.current_curation_activity&.status == 'published' # break once reached previous published
+        unchanged &&= res.files_unchanged?
+      end
+      resource.update_column(:file_view, false) if unchanged # if nothing changed between previous published and this, don't view same files again
```

Comment:
This logic looks correct, but it's always dangerous to have a lot of logic based on negative variables, like "unchanged". Can we change this to a positive "changed"?

Reply:
Ok, updated the logic here.